pkg/util/arm/deploy.go

```diff
@@ -28,13 +28,18 @@ func DeployTemplate(ctx context.Context, log *logrus.Entry, deployments features
 		},
 	})
 
+	if err == nil {
```

Comment:
What's this particular `if` statement here for? I think you called it out in the PR review meeting, but I forgot.

Reply:
I basically added a short-circuit here, mainly just to reproduce the flaky role assignment propagation issues in my local environment when running with a debugger attached. The rest of this function handles a `nil` err the way we would have wanted, and the function will correctly return `nil` upon completion, this just allows the function to exit early upon success rather than performing a bunch of error checks against a `nil` error.

Reply:
Ah, makes sense. As far as keeping it in the codebase goes, I see it as an unnecessary optimization that's confusing for someone else reading the code, so I'd recommend removing it. It's an opinion thing though, so no big deal either way. 🙂

Reply:
I'm happy to remove it, I'll leave this open for a bit for more thoughts though.

Reply:
I removed the short-circuit check.